alma/lib/Helpers/Admin/InsuranceHelper.php

```diff
@@ -216,4 +229,156 @@ public function saveConfigInsurance($config)
+                    }
+                }
+            } catch (InsuranceInitException $e) {
+                Logger::instance()->error('[Alma] The insurance product has not been created');
+            }
+        }
+    }
+
+    /**
+     * Create the default Insurance attribute group
+     * @return void
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    protected function createAttributeGroupIfNotExist()
+    {
+        $insuranceAttributeGroup = $this->attributeGroupRepository->getAttributeIdByName(
+            ConstantsHelper::ALMA_INSURANCE_ATTRIBUTE_NAME,
+            $this->context->language->id
+        );
+
+        if (!$insuranceAttributeGroup) {
+            try {
+                $attributeGroup = new \AttributeGroup();
+                $attributeGroup->group_type = 'select';
+                $attributeGroup->name = $this->createMultiLangField(utf8_encode(ConstantsHelper::ALMA_INSURANCE_ATTRIBUTE_NAME));
+                $attributeGroup->public_name = $this->createMultiLangField(utf8_encode(ConstantsHelper::ALMA_INSURANCE_ATTRIBUTE_PUBLIC_NAME));
+                $attributeGroup->add();
+            } catch (InsuranceInitException $e) {
+                Logger::instance()->error('[Alma] The insurance attribute group has not been created');
+            }
+        }
+    }
+
+    /**
+     * @param $idEntity
+     * @param $idImage
+     * @param $imgUrl
+     * @return bool
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    protected function uploadImage($idEntity, $idImage = null, $imgUrl)
+    {
+        $tmpFile = tempnam(_PS_TMP_IMG_DIR_, 'ps_import');
+        $watermarkTypes = explode(',', \Configuration::get('WATERMARK_TYPES'));
+        $imageObj = new \Image((int)$idImage);
+        $path = $imageObj->getPathForCreation();
+        $imgUrl = str_replace(' ', '%20', trim($imgUrl));
+        // Evaluate the memory required to resize the image: if it's too big we can't resize it.
+        if (!\ImageManager::checkImageMemoryLimit($imgUrl)) {
+            return false;
+        }
+        if (@copy($imgUrl, $tmpFile)) {
+            \ImageManager::resize($tmpFile, $path . '.jpg');
+            $imagesTypes = \ImageType::getImagesTypes('products');
+            foreach ($imagesTypes as $imageType) {
+                \ImageManager::resize($tmpFile, $path . '-' . stripslashes($imageType['name']) . '.jpg', $imageType['width'], $imageType['height']);
+                if (in_array($imageType['id_image_type'], $watermarkTypes)) {
+                    \Hook::exec('actionWatermark', array('id_image' => $idImage, 'id_product' => $idEntity));
+                }
+            }
+        } else {
+            unlink($tmpFile);
+            return false;
+        }
+        unlink($tmpFile);
+        return true;
+    }
+
+    /**
+     * @param $field
+     * @return array
+     */
+    protected function createMultiLangField($field)
```

Comment:
instead of this function, use the prestashop one \PrestaShop\PrestaShop\Adapter\Import\ImportDataFormatter::createMultiLangField()